### ISSUE TITLE
Support loading TIFF subIFD's

### ISSUE THREAD

**Issue by thany** [lovell/sharp#2557]
_What are you trying to achieve?_
I'm trying to load a DNG, and DNG's are saved in parts in TIFF subIFD's. Regardless, subIFD is part of the TIFF format, and there doesn't seem to be a way to specify a subIFD when loading a file/buffer, nor is there a way to know how many subIFD's a TIFF has.

Libvips [appears to support this](https://libvips.github.io/libvips/API/current/VipsForeignSave.html#vips-tiffload).

Also please note that subIFD is not the same as pages. The file I'm trying to read has 1 page, so the pages feature doesn't resolve this.

Also, I know very well that DNG isn't supported, but the issue is more general than that - subIFD is a TIFF standard, and DNG just *happens* to take advantage of it. Supporting this TIFF feature might be a step towards being able to read DNG with sharp - or at least decode the bits I need at a fairly awesome speed.

_Have you searched for similar feature requests?_
The word subIFD doesn't occur anywhere in this entire repo (well, it does now, but I mean before creating this issue).

_What would you expect the API to look like?_
Could be a property `subIFDs: Array<Buffer>` in the `metadata()` object, each of which can be passed back into a `sharp()` constructor. This would be in line with other metadata, which are also returned as binary blobs.

Another option would be to simply add `subIFDs: number` just to know how many there are. Then you could pass a subIFD index to the constructor like `sharp('foo.dng', { subIFD: 2 })`. This would be somewhat in line with the libvips API.

_What alternatives have you considered?_
A custom TIFF reader, but it's very slow and clunky. I'm sure lipvips can do much better.

_Is there a sample image that helps explain?_
Any ol' DNG will do, but I'll be happy to supply one, if need be.

**Comment by lovell** [comment]
Hello, yes, libvips supports this feature so adding it to sharp should be fairly straightforward.

We'll need to expose the `subifd` property as an option on sharp's constructor and ensure it is set when loading TIFF images, in very much the same way that `level` is exposed for OpenSlide approximately here:

https://github.com/lovell/sharp/blob/0dc325daa4a5b15bdbd1d6ff2375e0d18b4380c7/src/common.cc#L317

Happy to accept a PR for this, if you're able.

**Comment by jcupitt** [comment]
It's just like the `n-pages` property, so you should be able to copy-paste the code for that:

```
$ vips copy k2.jpg x.tif[pyramid,subifd]
$ vipsheader -a x.tif
x.tif: 1450x2048 uchar, 3 bands, srgb, tiffload
width: 1450
height: 2048
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 2.835
yres: 2.835
filename: x.tif
vips-loader: tiffload
n-subifds: 4
n-pages: 1
resolution-unit: in
orientation: 1
```

ie. five layers in this pyramid, with the main image as layer 0 and four subifds.

**Comment by lovell** [comment]
Thanks John. For anyone looking to expose this `subifds` metadata in sharp, a good place to start is to follow the trail of existing code for the `pages` property returned by [metadata](https://sharp.pixelplumbing.com/api-input#metadata), approximately here:

https://github.com/lovell/sharp/blob/0dc325daa4a5b15bdbd1d6ff2375e0d18b4380c7/src/metadata.cc#L62

**Comment by lovell** [comment]
Commit https://github.com/lovell/sharp/commit/43a085d1ae126016bd7c9be74881983554214394 adds `subifds` to the `metadata` response and accepts an optional `subifd` via the constructor.

https://sharp.pixelplumbing.com/api-input#metadata
https://sharp.pixelplumbing.com/api-constructor#parameters

This is available in v0.28.1.
